### PR TITLE
Query: Relational: Throw exception when translating Count with predicate on Grouping

### PR DESCRIPTION
**Pull request by smitpatel**
cc: @roji

### CODE REVIEW

File: test/EFCore.Sqlite.FunctionalTests/Query/GroupByQuerySqliteTest.cs [hunk 1]
```diff
@@ -15,5 +18,19 @@ public GroupByQuerySqliteTest(NorthwindQuerySqliteFixture<NoopModelCustomizer> f
             Fixture.TestSqlLoggerFactory.Clear();
             //Fixture.TestSqlLoggerFactory.SetTestOutputHelper(testOutputHelper);
         }
+
```
Comment (maumar) [inline test/EFCore.Sqlite.FunctionalTests/Query/GroupByQuerySqliteTest.cs]: nit: remove empty lines

File: src/EFCore.InMemory/Query/Internal/InMemoryExpressionTranslatingExpressionVisitor.cs [hunk 2]
```diff
@@ -298,23 +327,55 @@ MethodInfo getMethod()
...
-                            : method.MakeGenericMethod(typeof(ValueBuffer));
-
-                        return Expression.Call(method,
-                            groupByShaperExpression.GroupingParameter,
-                            selector);
+                    }
 
                     case nameof(Enumerable.Count):
+                    {
+                        var predicate = GetPredicate(methodCallExpression, groupByShaperExpression);
```
Comment (maumar) [inline src/EFCore.InMemory/Query/Internal/InMemoryExpressionTranslatingExpressionVisitor.cs]: the only difference is MethodInfo - DRY

Comment (smitpatel) [reply src/EFCore.InMemory/Query/Internal/InMemoryExpressionTranslatingExpressionVisitor.cs]: Done

**Comment by sebstr** [comment]
Hi @smitpatel 
Will this case be supported any time soon?

**Comment by roji** [comment]
Great!

**Comment by smitpatel** [comment]
@sebstr 
> Will this case be supported

Yes. See #11711

> Any time soon?

We haven't planned for release after 3.1 yet. So we don't know.

**Comment by AlekseyMartynov** [comment]
Hello @smitpatel 

I checked the recent daily build `3.1.0-preview3.19554.8`.
Still fails.

Query:
```c#
var x = context.Set<DataItem>()
    .GroupBy(obj => new {
        K1 = obj.Group1,
        K2 = obj.Group2
    })
    .OrderBy(g => g.Key.K1)
    .ThenBy(g => g.Key.K2)
    .Select(g => new {
        A1 = g.Count(obj => obj.Value != null),
        A2 = g.Select(obj => obj.Value != null ? 1 : 0).Sum()
    })
    .ToArray();
```

```c#
public class DataItem {
    public int ID { get; set; }
    public string Group1 { get; set; }
    public string Group2 { get; set; }
    public int? Value { get; set; }
}
```

Error:
```
  Message: 
    System.InvalidOperationException : The LINQ expression '(GroupByShaperExpression:
    KeySelector: new { 
        K1 = (s.Group1), 
        K2 = (s.Group2)
     }, 
    ElementSelector:(EntityShaperExpression: 
        EntityType: DataItem
        ValueBufferExpression: 
            (ProjectionBindingExpression: EmptyProjectionMember)
        IsNullable: False
    )
    )
        .Count(obj => obj.Value != null)' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to either AsEnumerable(), AsAsyncEnumerable(), ToList(), or ToListAsync(). See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
  Stack Trace: 
    RelationalSqlTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
    MethodCallExpression.Accept(ExpressionVisitor visitor)
    ExpressionVisitor.Visit(Expression node)
    RelationalSqlTranslatingExpressionVisitor.Translate(Expression expression)
    RelationalProjectionBindingExpressionVisitor.Visit(Expression expression)
    RelationalProjectionBindingExpressionVisitor.VisitNew(NewExpression newExpression)
    NewExpression.Accept(ExpressionVisitor visitor)
    ExpressionVisitor.Visit(Expression node)
    RelationalProjectionBindingExpressionVisitor.Visit(Expression expression)
    RelationalProjectionBindingExpressionVisitor.Translate(SelectExpression selectExpression, Expression expression)
    RelationalQueryableMethodTranslatingExpressionVisitor.TranslateSelect(ShapedQueryExpression source, LambdaExpression selector)
    QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
    RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
    MethodCallExpression.Accept(ExpressionVisitor visitor)
    ExpressionVisitor.Visit(Expression node)
    QueryCompilationContext.CreateQueryExecutor[TResult](Expression query)
    Database.CompileQuery[TResult](Expression query, Boolean async)
    QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async)
    <>c__DisplayClass9_0`1.<Execute>b__0()
    CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
    CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
    QueryCompiler.Execute[TResult](Expression query)
    EntityQueryProvider.Execute[TResult](Expression expression)
    EntityQueryable`1.GetEnumerator()
    LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
    EnumerableHelpers.ToArray[T](IEnumerable`1 source)
    Enumerable.ToArray[TSource](IEnumerable`1 source)
```

**Pull request merged**